### PR TITLE
Intercom - Add user info

### PR DESCRIPTION
**Pull request by adriangohjw**
## Problem

Currently intercom is implemented but does not take in user's information. However Intercom stores information in browser cookie (might be wiped out), thus conversations not preserved across devices and sessions

## Solution

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [x] No - this PR is backwards compatible

**Improvements**:

- pass in information into IntercomSDK after retrieving from `getMe()`
- extract into a standalone component to abstract away logics

### CODE REVIEW

File: apps/studio/src/components/Intercom.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+export const Intercom = () => {
+  const { me } = useMe()
+
+  if (env.NEXT_PUBLIC_INTERCOM_APP_ID) {
+    IntercomSDK({
+      app_id: env.NEXT_PUBLIC_INTERCOM_APP_ID,
+      user_id: me.id,
+      name: me.name,
```
Comment (dcshzj) [inline apps/studio/src/components/Intercom.tsx:17]: issue: I think we should be using `preferredName` here instead? Also, the `name` column can be empty, so we should fallback to the front part of the email address with replacements to spaces if it is not present.

Comment (adriangohjw) [reply apps/studio/src/components/Intercom.tsx:17]: actually, on this, why is there a need for both `preferredName` and `name` columns?

Also, i've looked at the DB, `name` are all filled but not `preferredName`, so it's safer to use `name` instead

Noted on the fallback on email, will do!

Comment (dcshzj) [reply apps/studio/src/components/Intercom.tsx:17]: Hmm the `preferredName` column was created specifically for Intercom actually, iirc it is to make the widget more personal to the user. I think it's just very low on our priority list to allow users to define their preferred names in Studio.

Comment (adriangohjw) [reply apps/studio/src/components/Intercom.tsx:17]: so intended use case was for name to be `Adrian Goh Jun Wei` and preferred name be `Adrian`?

Comment (dcshzj) [reply apps/studio/src/components/Intercom.tsx:17]: Yup, so the Intercom widget will say "Hi, Adrian" rather than the full name

Comment (adriangohjw) [reply apps/studio/src/components/Intercom.tsx:17]: okay lemme fallback from `preferredName` > `name` > `email` username

**Review by dcshzj — APPROVED** [review]

**Pull request merged**